File: Backend/src/controllers/developerScore.controller.js
const githubService = require("../services/github.service");

const asyncHandler = require("../middleware/asyncHandler");

const getDeveloperScore = asyncHandler(async (req, res) => {
  const { username } = req.params;

  /*
   * Validation
   */
  if (!username?.trim()) {
    return res.status(400).json({
      success: false,
      message: "Username is required.",
    });
  }

  const data = await githubService.getDeveloperScore(username);

  return res.status(200).json({
    success: true,
    data,
  });
});

module.exports = {
  getDeveloperScore,
};

File: Backend/app/controllers/developer_score_controller.py
"""
Developer score controller.
Port of controllers/developerScore.controller.js.
"""

from app.services.developer_score_service import get_developer_score


async def handle_get_developer_score(username: str) -> dict:
    """
    Handle GET /api/github/developer-score/:username

    Returns:
        { success: true, data: { score, level } }
    """
    if not username or not username.strip():
        raise ValueError("Username is required.")

    data = await get_developer_score(username)
    return {"success": True, "data": data}
